### PR TITLE
javacc 7.0.5

### PR DESCRIPTION
**Pull request by martinswanson**
- [Y] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [Y] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [Y] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [Y] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [Y] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/javacc.rb [hunk 1]
```diff
@@ -1,27 +1,27 @@
 class Javacc < Formula
-  desc "Java parser generator"
+  desc "The most popular parser generator for use with Java applications"
```
Comment (fxcoudert) [inline Formula/javacc.rb]: Let's avoid “most popular”. In fact, let's keep the previous description.

File: Formula/javacc.rb [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
 
   depends_on "ant" => :build
   depends_on :java
 
   def install
     system "ant"
-    libexec.install "target/javacc.jar"
-    doc.install Dir["www/doc/*"]
-    (share/"examples").install Dir["examples/*"]
+    (libexec/"lib").install "target/javacc-7.0.5.jar"
```
Comment (fxcoudert) [inline Formula/javacc.rb]: Why are you moving docs and examples? They were fine.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/javacc.rb (hunk 1 above)
Comment (martinswanson) [reply Formula/javacc.rb]: - We would like the strapline to be consistent with the new project website we are launching at https://javacc.github.io/javacc/
- The docs were out of date and have been completely rewritten in markdown for 7.0.5.
- Docs have moved from “doc” directory to “docs” per convention for hosting on GitHub Pages
- Docs, examples, and lib have been moved under libexec to be consistent with projects like Flink.
- Previous version 7.0.4 was created without consultation with the JavaCC project team and will not work with any version after 7.0.4.

Comment (martinswanson) [reply Formula/javacc.rb]: - We would like the strapline to be consistent with the new project website we are launching at https://javacc.github.io/javacc/
- The docs were out of date and have been completely rewritten in markdown for 7.0.5.
- Docs have moved from “doc” directory to “docs” per convention for hosting on GitHub Pages
- Docs, examples, and lib have been moved under libexec to be consistent with projects like Flink.
- Previous version 7.0.4 was created without consultation with the JavaCC project team and will not work with any version after 7.0.4.

File: Formula/javacc.rb (hunk 2 above)
Comment (fxcoudert) [inline Formula/javacc.rb]:
```suggestion
    (libexec/"lib").install "target/javacc-#{version}.jar"
```

File: Formula/javacc.rb [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
+    (libexec/"docs").install Dir["docs/*"]
+    (libexec/"examples").install Dir["examples/*"]
     %w[javacc jjdoc jjtree].each do |script|
       (bin/script).write <<~SH
         #!/bin/bash
-        exec java -classpath #{libexec/"javacc.jar"} #{script} "$@"
+        exec java -classpath #{libexec/"lib/javacc-7.0.5.jar"} #{script} "$@"
```
Comment (fxcoudert) [inline Formula/javacc.rb]:
```suggestion
        exec java -classpath #{libexec/"lib/javacc-#{version}.jar"} #{script} "$@"
```

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/javacc.rb (hunk 3 above)
Comment (martinswanson) [reply Formula/javacc.rb]: This is fine, but note it will not be backwards compatible for any release prior to 7.0.5.

**Comment by martinswanson** [comment]
What is the next step please? We want users to be able to install the latest version of JavaCC via Homebrew. We have accepted the updates to add version.

**Comment by martinswanson** [comment]
Thank you for suggesting the changes. I have accepted them and marked for review. I think it is ready for you to accept ?

**Comment by fxcoudert** [comment]
- The new `desc` is not ok
- Please keep documentation in `doc`
- Examples should be in `pkgshare` (not `share`, nor `libexec`)
- Why install jar files in `libexec/lib` rather than in `libexec`, I don't really see the point?

**Comment by martinswanson** [comment]
- Documentation must be in docs/ because we are using GitHub Pages to host the site. Otherwise we end up with docs/ in the project and doc/ in the distribution, which is not consistent. 
- We want to standardise the location of examples and .jars to follow the conventions of Apache Flink. 
- Happy to change the description.

Please note the previous version of the formula was created by someone not associated with the project and without consultation with the project team. The set up he implemented is now broken and not using the latest version. We do not want to be bound by this implementation going forward.

Best wishes,

Martin

File: Formula/javacc.rb (hunk 3 above)
Comment (fxcoudert) [reply Formula/javacc.rb]: We do not need backward compatibility, we only ship a single version at any time.

**Comment by fxcoudert** [comment]
> Otherwise we end up with docs/ in the project and doc/ in the distribution, which is not consistent

Documentation in all Homebrew formulas is installed in `doc`, which ends up being linked in the same place. So for consistency, as part of Homebrew's formula, we want to retain all docs in the same place. This is due to how Homebrew's symbolic link systems works.

> We want to standardise the location of examples and .jars to follow the conventions of Apache Flink

For jar files, we're fine with it. For examples, again `pkgshare` is Homebrew's standard install path.

> was created by someone not associated with the project and without consultation with the project team

That's the spirit of open source! We welcome contribution from upstream to our formulas, of course, but we'll try to keep things where Homebrew users expect them.

**Comment by BlueGone** [comment]
@martinswanson 
I am the one who first submitted this formula with #45064, as you said, without consultation with the javacc team. I was willing to do my best for to make javacc available and usable. The way I designed this formula was highly inspired by a PKGBUILD from [the javacc AUR PKGBUILD for Arch Linux](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=javacc), that I adapted to the standard path for Homebrew. It it the first formula I proposed for homebrew-core, so it may not be perfect, but I really did my best to make.
Now I would like to update the formula to version 7.0.5. Do you want to do it yourself, or should I do it ?

**Comment by martinswanson** [comment]
Hi Thomas

I already tried to push a version for 7.0.5 but the Homebrew team refused to accept the changes. Suggest you go ahead and upgrade it yourself.
Best wishes,
Martin

**Comment by SMillerDev** [comment]
A straightforward update would be accepted fairly quickly. 
Moving files to unexpected locations in the filesystem will encounter resistance from the people who maintain homebrew since you won't be here to help users figure it out, we are.

**Comment by BlueGone** [comment]
I just proposed a formula here : #48077

**Comment by fxcoudert** [comment]
Closing in favour of https://github.com/Homebrew/homebrew-core/pull/48077

**Pull request closed without merging**